pkg/controller/core/workload_controller.go

```diff
@@ -587,11 +588,31 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 		if err := r.cache.UpdateWorkload(oldWl, wlCopy); err != nil {
 			log.Error(err, "Updating workload in cache")
 		}
+		// This forces you to go through the scheduler to update PodSetAssignments if there's a difference between the
+		// worker group podSetAssignments.Count of the old and new workload
+		updatePsa := compareAdmissionPodSetAssignmentCount(oldWl.Status.Admission, wlCopy.Status.Admission)
+		if features.Enabled(features.DynamicallySizedJobs) && updatePsa && !r.queues.UpdateWorkload(oldWl, wlCopy) {
+			log.V(2).Info("Queue for updated workload didn't exist; ignoring for now")
```

Comment:
This check just adds a log message and doesn't change any behavior of the controller. Am I missing something?

Reply:
It's checking if the feature is enabled and if there's a diff in podset count, if so it calls `!r.queues.UpdateWorkload(oldWl, wlCopy)`, I can change the log message

Reply:
Oh I see, I think the readability can improve like this:
```
if features.Enabled(features.DynamicallySizedJobs) && compareAdmissionPodSetAssignmentCount(oldWl.Status.Admission, wlCopy.Status.Admission) {
  if !r.queues.UpdateWorkload(oldWl, wlCopy)  {
    log.V(2).Info("Queue for updated workload didn't exist; ignoring for now")
  }
}
```